### ISSUE TITLE
GraphQLNullTypeAttribute request

### ISSUE THREAD

**Issue by danny-zegel-zocdoc** [ChilliCream/graphql-platform#5980]
### Product

Hot Chocolate

### Is your feature request related to a problem?

With a nullable enabled solution I would like to have compile time enforcement on nullability without having error boundaries always tightly coupled with the nullability of my C# types.

In particular the issue I'm having is that I want my queries and mutations to be able to return a non-nullable type while still returning a 200 response if an exception is thrown.

e.g. given the query
```cs
public async Task<TheStuff> GetStuff(string stuffId) { ... }
```
the GQL schema generated is `stuff(stuffId: String!): TheStuff!` which results in a 500 response if the query throws an exception

The solution Iv'e come up with for now is to do
```cs
[GraphQLType(typeof(TheStuff))]
public async Task<TheStuff> GetStuff(string stuffId) { ... }
```
which results in the GQL schema `stuff(stuffId: String!): TheStuff` which returns a 200 if an exception is thrown.

However aside from this solution being awkward and unintuitive it's also imperfect because this code
```cs
[GraphQLType(typeof(List<Stuff>))]
public Task<List<Stuff>> GetStuffs(string stuffId) { ... }
```
results in the GQL schema `stuffs(stuffId: String!): [TheStuff]` but I would want it to be `stuffs(stuffId: String!): [TheStuff!]`

### The solution you'd like

I think a simple solution to this problem would be create `GraphQLNullTypeAttribute` similar to the `GraphQLNonNullTypeAttribute`.

I understand that the intention of `GraphQLNonNullTypeAttribute` was likely to support solutions where null enforcement is not enabled but I think using  my proposed `GraphQLNullTypeAttribute` in a nullable enables solution would provide a lot of flexibility so that the C# type and the GQL schemas could diverge where appropriate.

I could also see this being useful to control GQL error boundaries deeper in the GQL schema while maintaining compiler time enforcement on on the desired C# types.

**Comment by michaelstaib** [comment]
@danny-zegel-zocdoc yes but its not so elegant.... btw ... did you know that you also can do `[GraphQLType<TheStuff>]`. 
Anyway we still have this one on the backlog for 14.
